### PR TITLE
Added 'slow' command.

### PR DESCRIPTION
**Pull request by bartoszj**
This command slows down animations. Works like 'Toggle Slow Animations' from iOS Simulator but this command works also on the device and user is able to select animation speed.

### CODE REVIEW

**Comment by kastiglione** [comment]
Thank you @bartoszj :+1: 

Would you mind signing Facebook's Contributor License Agreement? If you have any questions I'd be happy to help answer. https://code.facebook.com/cla

**Comment by bartoszj** [comment]
Done.

File: commands/FBDisplayCommands.py [hunk 1]
```diff
@@ -145,3 +146,17 @@ def args(self):
 
   def run(self, args, options):
     viewHelpers.setViewHidden(args[0], True)
+
+
+class FBSlowAnimation(fb.FBCommand):
+  def name(self):
+    return 'slow'
```
Comment (kastiglione) [inline commands/FBDisplayCommands.py]: What do you think about calling this `slowanim`?

File: commands/FBDisplayCommands.py [hunk 2]
```diff
@@ -145,3 +146,17 @@ def args(self):
...
+
+  def description(self):
+    return 'Slows down animations. Works on the iOS Simulator and a device.'
+
+  def args(self):
+    return [ fb.FBCommandArgument(arg='speed', type='float', default='1', help='Animation speed.') ]
+
+  def run(self, args, option):
+    lldb.debugger.HandleCommand('expr [(CALayer *)[[[[UIApplication sharedApplication] windows] objectAtIndex:0] layer] setSpeed:(CGFloat)%s]' % (args[0]))
```
Comment (kastiglione) [inline commands/FBDisplayCommands.py]: Can we run this for all windows instead of just the one at index 0?

**Comment by kastiglione** [comment]
@bartoszj Thanks for submitting this! I have a couple comments above.

What do you think about adding two commands, one to turn this on, one to turn if off. All of our other commands work this way, so we should be consistent unless there's a good reason not to be. If we have two commands, they can both have a default speed, with the undo command having a speed of 1.

**Comment by arigrant** [comment]
Isn't this the same as 'slowanim' and 'unslowanim' or have we not merged that, @kastiglione ?

**Comment by bartoszj** [comment]
@kastiglione I agree with two commands. `slowanim` and `unslowanim` would be OK? Also I'll check slow animation on all windows.

**Comment by kastiglione** [comment]
Great! :bow:

**Comment by bartoszj** [comment]
Updated.

File: fblldbviewhelpers.py [hunk 3]
```diff
@@ -87,3 +87,6 @@ def upwardsRecursiveDescription(view, maxDepth=0):
     currentPrefix += "   | "
 
   return builder
+
+def slowAnimation(speed=1):
+  lldb.debugger.HandleCommand('expr -- for (UIWindow *$w in (NSArray *)[[UIApplication sharedApplication] windows]) { [[$w layer] setSpeed:(CGFloat)%s]; }' % (speed))
```
Comment (kastiglione) [inline fblldbviewhelpers.py]: This is awesome!

For what it's worth, another way that might have worked is to use `NSArray`'s custom implementation of `-setValue:forKey:`:

```objc
[[[UIApplication sharedApplication] windows] setValue:@(speed) forKey:@"speed"]
```

Comment (kastiglione) [inline fblldbviewhelpers.py]: Where allowable, can we change the `XYClass *` casts to simply `id`?

File: commands/FBDisplayCommands.py [hunk 4]
```diff
@@ -145,3 +147,28 @@ def args(self):
...
+    return 'slowanim'
+
+  def description(self):
+    return 'Slows down animations. Works on the iOS Simulator and a device.'
+
+  def args(self):
+    return [ fb.FBCommandArgument(arg='speed', type='float', help='Animation speed.') ]
+
+  def run(self, args, option):
+    viewHelpers.slowAnimation(args[0])
```
Comment (kastiglione) [inline commands/FBDisplayCommands.py]: Can we provide a default? If I call `slowanim`, maybe set it to 0.25? Perhaps we can dig up the speed the simulator uses, but that's hardly worth spending much time on.

**Comment by kastiglione** [comment]
Thanks @bartoszj! I left a couple comments, the main one is about giving `slowanim` a default speed value so that it could be typed without having to provide an explicit speed.

**Comment by bartoszj** [comment]
I've found that iOS Simulator is using 0.1 as speed value and I've set it in the`slowanim` command.

**Comment by kastiglione** [comment]
:boom: 

Looks good. I'll checkout this branch and give it a try and merge if all goes well. Thanks!

**Comment by kastiglione** [comment]
Thanks for this contribution! And thanks for being patient and awesome :+1:

**Pull request merged**